common/service/config/config.go

```diff
@@ -189,6 +191,16 @@ type (
 		Client ClientTLS `yaml:"client"`
 	}
 
+	// CertExpirationValidation contains settings for periodic checks of TLS certificate expiration
+	CertExpirationValidation struct {
+		// Log warnings for certificates expiring during this period of time from now
+		WarningPeriod time.Duration `yaml:"warningPeriod"`
```

Comment:
[Nit] 1) call this "duration" instead of "period" 2) the comment needs to be re-worded, it's hard to grok. (maybe, log warnings for certificates that will expire within this duration)

Reply:
How about "WarningWindow" and "ErrorWindow" instead? I'm slightly concerned that "WarningDuration" sounds as if the warning would be up for that duration.